src/monitors.ts

```diff
@@ -0,0 +1,180 @@
+import { SERVERLESS_MONITORS } from "./serverless_monitors";
+import { updateMonitor, createMonitor, deleteMonitor, getExistingMonitors } from "./monitor-api-requests";
+
+export interface MonitorParams {
+  [key: string]: any;
+}
+export interface Monitor {
+  [key: string]: MonitorParams;
+}
+
+/**
+ * Adds the appropriate tags and required parameters that will be passed as part of the request body for creating and updating monitors
+ * @param monitor - the Monitor object that is defined in the serverless.yml file
+ * @param cloudFormationStackId - the Cloud Formation Stack ID
+ * @param service - the Service
+ * @param env  - the Environment
+ * @returns valid monitor parameters
+ */
+export function buildMonitorParams(monitor: Monitor, cloudFormationStackId: string, service: string, env: string) {
+  const serverlessMonitorId = Object.keys(monitor)[0];
+
+  if (!monitor[serverlessMonitorId]) {
+    monitor[serverlessMonitorId] = {};
+  }
+
+  const monitorParams = { ...monitor[serverlessMonitorId] };
+
+  if (!monitorParams.tags) {
+    monitorParams.tags = [];
+  }
+  if (!monitorParams.options) {
+    monitorParams.options = {};
+  }
+  if (monitorParams.type === undefined) {
+    monitorParams.type = "metric alert";
+  }
+
+  monitorParams.tags = [
+    ...monitorParams.tags,
+    "serverless_monitor_type:single_function",
+    `serverless_monitor_id:${serverlessMonitorId}`,
+    `aws_cloudformation_stack-id:${cloudFormationStackId}`,
+    "created_by:dd_sls_plugin",
+    `env:${env}`,
+    `service:${service}`,
+  ];
+
+  if (checkIfServerlessMonitor(serverlessMonitorId)) {
+    monitorParams.query = SERVERLESS_MONITORS[serverlessMonitorId].query(cloudFormationStackId);
+
+    if (!monitorParams.message) {
+      monitorParams.message = SERVERLESS_MONITORS[serverlessMonitorId].message;
+    }
+    if (!monitorParams.name) {
+      monitorParams.name = SERVERLESS_MONITORS[serverlessMonitorId].name;
+    }
+  }
+  return monitorParams;
+}
+
+/**
+ * Checks to see if the given monitor is a serverless recommended monitor
+ * @param serverlessMonitorId - Unique ID string defined for each serverless monitor
+ * @returns true if a given monitor is a serverless recommended monitor
+ */
+function checkIfServerlessMonitor(serverlessMonitorId: string) {
```

Comment:
I would rename this to `checkIfRecommendedMonitor` -- it seems to me that all monitors (custom or recommended) are equally serverless, and what distinguishes these are not that they are serverless but that they are recommended.